presto-accumulo/src/test/java/com/facebook/presto/accumulo/index/TestIndexer.java

```diff
@@ -0,0 +1,222 @@
+        assertKeyValuePair(iter.next(), "def".getBytes(), "cf_arr", "___card___", "1");
+        assertKeyValuePair(iter.next(), "ghi".getBytes(), "cf_arr", "___card___", "1");
+        assertFalse(iter.hasNext());
+
+        scan.close();
+
+        indexer.index(m2);
+        indexer.close();
+
+        scan = conn.createScanner(table.getIndexTableName(), auths);
+        scan.setRange(new Range());
+        iter = scan.iterator();
+        assertKeyValuePair(iter.next(), AGE_VALUE, "cf_age", "row1", "");
+        assertKeyValuePair(iter.next(), AGE_VALUE, "cf_age", "row2", "");
+        assertKeyValuePair(iter.next(), "abc".getBytes(), "cf_arr", "row1", "");
+        assertKeyValuePair(iter.next(), "abc".getBytes(), "cf_arr", "row2", "");
+        assertKeyValuePair(iter.next(), M1_FNAME_VALUE, "cf_firstname", "row1", "");
+        assertKeyValuePair(iter.next(), M2_FNAME_VALUE, "cf_firstname", "row2", "");
+        assertKeyValuePair(iter.next(), "def".getBytes(), "cf_arr", "row1", "");
+        assertKeyValuePair(iter.next(), "ghi".getBytes(), "cf_arr", "row1", "");
+        assertKeyValuePair(iter.next(), "ghi".getBytes(), "cf_arr", "row2", "");
+        assertKeyValuePair(iter.next(), "mno".getBytes(), "cf_arr", "row2", "");
+        assertFalse(iter.hasNext());
+
+        scan.close();
+
+        scan = conn.createScanner(table.getMetricsTableName(), auths);
+        scan.setRange(new Range());
+
+        iter = scan.iterator();
+        assertKeyValuePair(iter.next(), AGE_VALUE, "cf_age", "___card___", "2");
+        assertKeyValuePair(iter.next(), Indexer.METRICS_TABLE_ROW_ID.array(), "___rows___",
+                "___card___", "2");
+        assertKeyValuePair(iter.next(), Indexer.METRICS_TABLE_ROW_ID.array(), "___rows___",
+                "___first_row___", "row1");
+        assertKeyValuePair(iter.next(), Indexer.METRICS_TABLE_ROW_ID.array(), "___rows___",
+                "___last_row___", "row2");
+        assertKeyValuePair(iter.next(), "abc".getBytes(), "cf_arr", "___card___", "2");
+        assertKeyValuePair(iter.next(), M1_FNAME_VALUE, "cf_firstname", "___card___", "1");
+        assertKeyValuePair(iter.next(), M2_FNAME_VALUE, "cf_firstname", "___card___", "1");
+        assertKeyValuePair(iter.next(), "def".getBytes(), "cf_arr", "___card___", "1");
+        assertKeyValuePair(iter.next(), "ghi".getBytes(), "cf_arr", "___card___", "2");
+        assertKeyValuePair(iter.next(), "mno".getBytes(), "cf_arr", "___card___", "1");
+        assertFalse(iter.hasNext());
+
+        scan.close();
+    }
+
+    private void assertKeyValuePair(Entry<Key, Value> e, byte[] row, String cf, String cq,
+            String value)
+    {
+        System.out.println(e);
+        assertEquals(row, e.getKey().getRow().copyBytes());
+        assertEquals(cf, e.getKey().getColumnFamily().toString());
+        assertEquals(cq, e.getKey().getColumnQualifier().toString());
+        assertEquals(value, new String(e.getValue().get()));
```

Comment:
Shouldn't you define charset then? Is JVMs default correct here? We use `UTF-8` throughout Presto.